tests/toranj/cli/test-706-multi-radio-trel-peer-addr-port-change-discovery.py

```diff
@@ -0,0 +1,150 @@
+from cli import verify_within
+import cli
+import time
+
+# -----------------------------------------------------------------------------------------------------------------------
+# Test description: This test validates the mechanism to discover and update the TREL peer info when processing
+# a received TREL packet from a peer (when peer IPv6 address or port number gets changes).
+#
+
+test_name = __file__[:-3] if __file__.endswith('.py') else __file__
+print('-' * 120)
+print('Starting \'{}\''.format(test_name))
+
+# -----------------------------------------------------------------------------------------------------------------------
+# Creating `cli.Node` instances
+
+speedup = 10
+cli.Node.set_time_speedup_factor(speedup)
+
+r1 = cli.Node(cli.RADIO_15_4_TREL)
+r2 = cli.Node(cli.RADIO_15_4_TREL)
+c1 = cli.Node(cli.RADIO_15_4_TREL)
+
+# -----------------------------------------------------------------------------------------------------------------------
+# Build network topology
+
+r1.form("trel-peer-disc")
+c1.join(r1, cli.JOIN_TYPE_REED)
+r2.join(r1)
+
+verify(r1.get_state() == 'leader')
+verify(r2.get_state() == 'router')
+verify(c1.get_state() == 'child')
+
+verify(r1.multiradio_get_radios() == '[15.4, TREL]')
+verify(r2.multiradio_get_radios() == '[15.4, TREL]')
+verify(c1.multiradio_get_radios() == '[15.4, TREL]')
+
+# -----------------------------------------------------------------------------------------------------------------------
+# Test Implementation
+
+
+def check_trel_peers(node, peer_nodes):
+    # Validate that `node` has discovered `peer_nodes` as its TREL peers
+    # and validate that the correct TREL socket address is discovered
+    # and used for each.
+    peers = node.trel_get_peers()
+    verify(len(peers) == len(peer_nodes))
+    peers_ext_addrs = [nd.get_ext_addr() for nd in peer_nodes]
+    for peer in peers:
+        for peer_node in peer_nodes:
+            if peer['Ext MAC Address'] == peer_node.get_ext_addr():
+                verify(peer_node.trel_test_get_sock_addr() == peer['IPv6 Socket Address'])
+                break
+        else:
+            verify(False)  # Did not find peer in `peer_nodes`
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# Check that all nodes see each other as TREL peers with the correct
+# TREL socket address.
+
+check_trel_peers(r1, [r2, c1])
+check_trel_peers(r2, [r1, c1])
+check_trel_peers(c1, [r1, r2])
+
+verify(int(r1.trel_test_get_notify_addr_counter()) == 0)
+verify(int(r2.trel_test_get_notify_addr_counter()) == 0)
+verify(int(c1.trel_test_get_notify_addr_counter()) == 0)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# Use test CLI command to force `r2` to change its TREL socket
+# address.
+
+time.sleep(10 / speedup)
+
+old_sock_addr = r2.trel_test_get_sock_addr()
+r2.trel_test_change_sock_addr()
```

Comment:
Yes, the socket port change is not explicitly checked in the test (I had to think about how to add the test for this and ended up adding new test-specific CLI commands to trigger an address change under the `simulation` platform).

Since the OT code treats the address and port as a combined `Ip6::SockAddr` instance, the same code would apply (we perform all tracking and comparison logic with `Ip6::SockAddr`, which encapsulates an `Ip6::Address` along with a port number).

I will see if it is easy to add a new test command to change the port.

Reply:
I added new CLI test command to trigger port change so we now cover it in test.